### PR TITLE
spdlog: 1.8.5 -> 1.9.2

### PR DESCRIPTION
**Pull request by tobim**
###### Things done
The new version is [incompatible with fmt-7](https://github.com/gabime/spdlog/issues/2050), so I override `fmt` for packages that use both. I built the reverse dependencies except the kodi and python stuff manually, but didn't verify correct functionality. Given the purpose of this library it is unlikely that it introduces runtime breakage.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134609` at 7321b0cb run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>23 packages skipped due to time constraints:</summary>
  <ul>
    <li>far2l</li>
    <li>gerbera</li>
    <li>hal-hardware-analyzer</li>
    <li>kodi (kodiPackages.kodi)</li>
    <li>kodi-gbm</li>
    <li>kodi-wayland</li>
    <li>kodiPackages.a4ksubtitles</li>
    <li>kodiPackages.inputstream-adaptive</li>
    <li>kodiPackages.inputstream-ffmpegdirect</li>
    <li>kodiPackages.inputstream-rtmp</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>16 packages built successfully:</summary>
  <ul>
    <li>bear</li>
    <li>cherrytree</li>
    <li>mtxclient</li>
    <li>pc-ble-driver</li>
    <li>python38Packages.pc-ble-driver-py</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-aer</li>
    <li>python38Packages.qiskit-aqua</li>
    <li>python38Packages.qiskit-ibmq-provider</li>
    <li>python38Packages.qiskit-ignis</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-aer</li>
    <li>python39Packages.qiskit-aqua</li>
    <li>python39Packages.qiskit-ibmq-provider</li>
    <li>python39Packages.qiskit-ignis</li>
    <li>spdlog (spdlog_1)</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4540,6 +4542,7 @@ with pkgs;
 
   vowpal-wabbit = callPackage ../applications/science/machine-learning/vowpal-wabbit {
     boost = boost17x;
+    fmt = fmt_8;
```
Comment (kvtb) [inline pkgs/top-level/all-packages.nix:4545]: maybe just `propagatedBuilldInputs = [ fmt_8 ]` in `spdlog_1` instead of populating `fmt_8` so many times ?

**Comment by kvtb** [comment]
also `pkgsStatic.spdlog_1` is broken

it needs something like
```diff
-        "-DSPDLOG_BUILD_SHARED=${if stdenv.hostPlatform.isStatic then "OFF" else "ON"}"
-        "-DSPDLOG_BUILD_STATIC=${if stdenv.hostPlatform.isStatic then "ON" else "OFF"}"
+        "-DBUILD_SHARED_LIBS=${if stdenv.hostPlatform.isStatic then "OFF" else "ON"}"
```

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/pull/145006

**Pull request closed without merging**